Controllers/TagController.cs

```diff
@@ -45,4 +45,28 @@ public IActionResult CreateTag(TagCreateDTO newTag)
         _dbContext.SaveChanges();
         return Created($"/api/tag/{createdTag.Id}", createdTag);
     }
+
+    [HttpGet("{id}")]
+    [Authorize]
+    public IActionResult GetTagById(int id)
+    {
+        return Ok(_dbContext.Tags
+        .Where(t => t.Id == id)
+        .Select(t => new TagNoNavDTO
+        {
+            Id = t.Id,
+            Name = t.Name
+        }).SingleOrDefault());
+    }
+
+    [HttpPut("{id}")]
+    [Authorize]
+    public IActionResult UpdateATag(Tag newTag, int id)
```

Comment:
Is there a reason you're not using a DTO, such as `TagUpdateDTO` for submitting updates?